src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyWithoutShouldStartLine.java

```diff
@@ -0,0 +1,63 @@
+package com.puppycrawl.tools.checkstyle.checks.blocks.rightcurly;
+
+public class InputRightCurlyWithoutShouldStartLine {
+
+    private int a;
+    private static int b;
+    {  a = 2; }  //violation if option=ALONE
+    static { b = 3; }  //violation if option=ALONE
+
+    void method1() {
+        Thread t = new Thread() {@Override public void run() {
+            int a; int b;} //violation
+        };
+    }
+
+    void method2(java.util.HashSet<String> set) {
+        java.util.Map<String, String> map1 = new java.util.LinkedHashMap<String, String>() {{
+            put("Hello", "World");
+            put("first", "second");
+            put("polygene", "lubricants");
+            put("alpha", "betical");}  //violation
```

Comment:
we need to skip validation of this expression blocks, as validation will be done by special new Check.
It is better to do no validation, rather than bothering user with false-positives.

Reply:
@romani It sounds like you are agreeing with my comment at https://github.com/checkstyle/checkstyle/issues/4057#issuecomment-398785395 ?

Reply:
I changing my mind .... in this particular case - there should be violation then user start to play with location of "}".
I am ok to allow only mode `{{ }}`  when braces on STATIC_INIT is the only why OBJ_BLOCK has.

@rnveach , are you ok with this ?

Reply:
Discussion is in other issue.

> allow only mode {{ }} when braces on STATIC_INIT 

This scenario is INSTANCE_INIT, not STATIC_INIT.

Reply:
Sorry, I meant to type INSTANCE_INIT.

Reply:
@romani @rnveach I also agree that we should validate inner `INSTANCE_INIT` rcurly in this check itself. The outer rcurly should pertaining to the anon class is the one that gets validated in the new check. So, this violation stays then right ?

Reply:
> So, this violation stays then right ?

yes.
@rnveach , please share your position on this case.

Reply:
I shared my position in issue. If someone likes writting `{{ }}` in anonymous class but `{\n}\n` in class they will come and say so when check changes. So you can go forward with how you want it.

Reply:
ok, lets finish this issue.
Lets have no violations on `{{ }}` (when INSTANCE_INIT is completey(without any symbols/spaces/.... in OBJ_BLOCK)). We should mention this in xdoc.
Any other cases should be violated in `alone` mode.

Reply:
This violation stays. No changes made.